userspace/libsinsp/filterchecks.cpp

```diff
@@ -2218,6 +2233,11 @@ bool sinsp_filter_check_thread::compare_full_aname(sinsp_evt *evt)
 	//
 	// No id specified, search in all of the ancestors
 	//
+	if(mt->detect_parent_state_loop())
```

Comment:
Could we integrate the loop detection algorithm in the traversal? In this way it's not necessary to traverse the entire "parent_thread" chain every time. I think the same is true also for apid and sname checks.